### DISCUSSION TITLE
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url:

### DISCUSSION THREAD

**Discussion by gaussiangit** [open-webui/open-webui#4780] [General]
Following docker logs!.

>    raise LocalEntryNotFoundError(
> huggingface_hub.utils._errors.LocalEntryNotFoundError: Cannot find an appropriate cached snapshot folder for the specified revision on the local disk and outgoing traffic has been disabled. To enable repo look-ups and downloads online, pass 'local_files_only=False' as input.
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_errors.py", line 304, in hf_raise_for_status
> /app
>     response.raise_for_status()
>   File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1024, in raise_for_status
>     raise HTTPError(http_error_msg, response=self)
> requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://huggingface.co/sentence-transformers/ollama/resolve/main/config.json
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/transformers/utils/hub.py", line 402, in cached_file
>     resolved_file = hf_hub_download(
>                     ^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
>     return fn(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1221, in hf_hub_download
>     return _hf_hub_download_to_cache_dir(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1325, in _hf_hub_download_to_cache_dir
>     _raise_on_head_call_error(head_call_error, force_download, local_files_only)
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1823, in _raise_on_head_call_error
>     raise head_call_error
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1722, in _get_metadata_or_catch_error
>     metadata = get_hf_file_metadata(url=url, proxies=proxies, timeout=etag_timeout, headers=headers)
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
>     return fn(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 1645, in get_hf_file_metadata
>     r = _request_wrapper(
>         ^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 372, in _request_wrapper
>     response = _request_wrapper(
>                ^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py", line 396, in _request_wrapper
>     hf_raise_for_status(response)
>   File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_errors.py", line 352, in hf_raise_for_status
>     raise RepositoryNotFoundError(message, response) from e
> huggingface_hub.utils._errors.RepositoryNotFoundError: 401 Client Error. (Request ID: Root=1-66c5ab4c-4cc4111f539ffbf64c3e494d;70c6060c-538b-45a2-89e6-be8417484ddd)
> 
> Repository Not Found for url: https://huggingface.co/sentence-transformers/ollama/resolve/main/config.json.
> Please make sure you specified the correct `repo_id` and `repo_type`.
> If you are trying to access a private or gated repo, make sure you are authenticated.
> Invalid username or password.
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/usr/local/bin/uvicorn", line 8, in <module>
>     sys.exit(main())
>              ^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1157, in __call__
>     return self.main(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1078, in main
>     rv = self.invoke(ctx)
>          ^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/click/core.py", line 1434, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/click/core.py", line 783, in invoke
>     return __callback(*args, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/main.py", line 410, in main
>     run(
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/main.py", line 578, in run
>     server.run()
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 61, in run
>     return asyncio.run(self.serve(sockets=sockets))
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/asyncio/runners.py", line 190, in run
>     return runner.run(main)
>            ^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
>     return self._loop.run_until_complete(task)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "uvloop/loop.pyx", line 1517, in uvloop.loop.Loop.run_until_complete
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/server.py", line 68, in serve
>     config.load()
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/config.py", line 473, in load
>     self.loaded_app = import_from_string(self.app)
>                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/uvicorn/importer.py", line 21, in import_from_string
>     module = importlib.import_module(module_str)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 940, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/app/backend/main.py", line 46, in <module>
>     from apps.rag.main import app as rag_app
>   File "/app/backend/apps/rag/main.py", line 228, in <module>
>     update_reranking_model(
>   File "/app/backend/apps/rag/main.py", line 214, in update_reranking_model
>     app.state.sentence_transformer_rf = sentence_transformers.CrossEncoder(
>                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/sentence_transformers/cross_encoder/CrossEncoder.py", line 74, in __init__
>     self.config = AutoConfig.from_pretrained(
>                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/transformers/models/auto/configuration_auto.py", line 965, in from_pretrained
>     config_dict, unused_kwargs = PretrainedConfig.get_config_dict(pretrained_model_name_or_path, **kwargs)
>                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/transformers/configuration_utils.py", line 632, in get_config_dict
>     config_dict, kwargs = cls._get_config_dict(pretrained_model_name_or_path, **kwargs)
>                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/transformers/configuration_utils.py", line 689, in _get_config_dict
>     resolved_config_file = cached_file(
>                            ^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/transformers/utils/hub.py", line 425, in cached_file
>     raise EnvironmentError(
> OSError: sentence-transformers/ollama is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
> If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>`

**Comment by cl93a** [comment]
Similar issue here. How was this resolved?

**Comment by rossm-mf** [comment]
someone said they found it here but I am not even sure what is the fix
https://github.com/open-webui/open-webui/discussions/5357
